src/components/Slide/CanvasHandler.ts

```diff
@@ -93,16 +95,17 @@ export class CanvasHandler {
 
     group.on("moving", () => {
       this.canvas?.forEachObject((element) => {
-        if (!isEqual(element, group) && areObjectsNear(element, group)) {
-          text.set(
-            "text",
-            (element as fabric.Object & { customData: string }).customData
-          );
-
-          setTimeout(() => {
-            text.set("text", "");
-          }, 700);
-        }
+        if (!(element as FabricObject).customData) return;
+
+        if (isEqual(element, group)) return;
+
+        if (!areObjectsNear(element, group)) return;
+
+        text.set("text", (element as FabricObject).customData);
+
+        setTimeout(() => {
+          text.set("text", "");
```

Comment:
why use settimeout?

Reply:
I felt like having the sensed data from the proximity sensor is better removed